bindata/cloud-network-config-controller/managed/controller.yaml

```diff
@@ -188,13 +188,15 @@ spec:
         - name: "NO_PROXY"
           value: "{{ .NO_PROXY}}"
 {{ end }}
-{{- if not (eq .AzureManagedClientID "")}}
+{{- if not (eq .AzureManagedSecretProviderClass "")}}
```

Comment:
Just to confirm, are you confident moving the check to `AzureManagedSecretProviderClass`  will always work for the old `AzureManagedClientID` approach?

Reply:
Yeah, it is still used in both the new and old auth methods we were asked to implement.